libs/designer/src/lib/core/utils/outputs.ts

```diff
@@ -342,17 +349,26 @@ export const getUpdatedManifestForSchemaDependency = (manifest: OperationManifes
 
       const currentSchemaValue = getObjectPropertyValue(updatedManifest.properties.outputs, outputLocation);
 
+      const inputSchema = updatedManifest.properties?.inputs?.properties?.schema;
+      const isRequestApiConnectionTrigger =
+        'x-ms-editor-options' in inputSchema ? inputSchema['x-ms-editor-options'].isRequestApiConnectionTrigger : undefined;
+
+      let schemaValue: Schema;
+      let shouldMerge: boolean;
       // if schema contains static object returned from RP, merge the current schema value and new schema value
-      if (schemaToReplace && 'rows' in schemaToReplace) {
-        if (!Object.keys(schemaToReplace.rows.items.properties) !== undefined && 'rows' in currentSchemaValue) {
-          safeSetObjectPropertyValue(
-            updatedManifest.properties.outputs,
-            outputLocation,
-            { ...currentSchemaValue, ...schemaToReplace },
-            true
-          );
+      if (isRequestApiConnectionTrigger && schemaToReplace && 'rows' in schemaToReplace) {
+        if ('rows' in currentSchemaValue) {
+          schemaValue = { ...currentSchemaValue, ...schemaToReplace };
+          shouldMerge = true;
```

Comment:
Suppose someone added parameters using the floating editor. How does deleting the parameter work if this is always merging with the existing schema.

Reply:
`schemaToReplace` in the case of a floatingActionMenu is the value that's added in the rootObject. Incase there is a delete, the rootObject is updated accordingly. Incase the rootObject gets empty, the `schemaToReplace` would be an empty rootObject (Incase of a hybridTrigger, the rootObject would contain the 'rows' object, for manualTrigger there won't be). The merge would be a. if the currentSchemaValue is empty, then nothing to merge, b. if the currentSchemValue has a value then merge with emptyRootObject

Reply:
I don't understand what you are saying, mostly because I don't know the terminology being used.

I'm picturing the currentSchemaValue containing something like:
```
{
"type": "object",
"properties": {
    "ID": {
        "type": "integer",
        "format": "int64",
        "description": "File Identifier"
    },
    "fileName": {
        "type": "string",
        "description": "File Name"
    }
}
```
and schemaToReplace is (after delete):
```
{
"type": "object",
"properties": {
    "ID": {
        "type": "integer",
        "format": "int64",
        "description": "File Identifier"
    }
}
```

will the resulting merge be without `fileName`?

Is this scenario possible and is somehow handled, that's my concern here.

Reply:
This scenario after delete will be merged and final result will be the following:
```{
"type": "object",
"properties": {
    "ID": {
        "type": "integer",
        "format": "int64",
        "description": "File Identifier"
    }
}```

I have tested this scenario

Reply:
ok